bioblend/_tests/TestGalaxyDatasets.py

```diff
@@ -57,15 +62,124 @@ def test_get_datasets(self):
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, extension='txt')
+        self.assertEqual(len(datasets), 2)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, extension='bam')
+        self.assertEqual(datasets, [])
+
+    @test_util.skip_unless_galaxy('release_20.05')
+    def test_get_datasets_state(self):
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, state='ok')
+        self.assertEqual(len(datasets), 2)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, state='queued')
+        self.assertEqual(datasets, [])
+        with self.assertRaises(ConnectionError):
+            self.gi.datasets.get_datasets(history_id=self.history_id, state='nonexistent_state')
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, state=['ok', 'queued'])
+        self.assertEqual(len(datasets), 2)
+
+    @test_util.skip_unless_galaxy('release_20.05')
+    def test_get_datasets_visible(self):
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, visible=True)
+        self.assertEqual(len(datasets), 2)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, visible=False)
+        self.assertEqual(len(datasets), 0)
+
+    @test_util.skip_unless_galaxy('release_19.05')
+    def test_get_datasets_ordering(self):
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, order='create_time-dsc')
+        self.assertEqual(datasets[0]['id'], self.dataset_id2)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, order='create_time-asc')
+        self.assertEqual(datasets[0]['id'], self.dataset_id)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, order='hid-dsc')
+        self.assertEqual(datasets[0]['id'], self.dataset_id2)
+        datasets = self.gi.datasets.get_datasets(history_id=self.history_id, order='hid-asc')
+        self.assertEqual(datasets[0]['id'], self.dataset_id)
+
+    @test_util.skip_unless_galaxy('release_19.05')
+    def test_get_datasets_deleted(self):
+        deleted_datasets = self.gi.datasets.get_datasets(history_id=self.history_id, deleted=True)
+        self.assertEqual(deleted_datasets, [])
+        self.gi.histories.delete_dataset(self.history_id, self.dataset_id2)
+        deleted_datasets = self.gi.datasets.get_datasets(history_id=self.history_id, deleted=True)
+        self.assertEqual(len(deleted_datasets), 1)
+        purged_datasets = self.gi.datasets.get_datasets(history_id=self.history_id, purged=True)
+        self.assertEqual(purged_datasets, [])
+        self.gi.histories.delete_dataset(self.history_id, self.dataset_id2, purge=True)
+        purged_datasets = self.gi.datasets.get_datasets(history_id=self.history_id, purged=True)
+        self.assertEqual(len(purged_datasets), 1)
+        self.tearDown()
+        self.setUp()
```

Comment:
I doubt that does anything to the history, I will add suggestions for what I meant.
```suggestion
```